### ISSUE TITLE
Unhelpful error when adding new asset to an existing software-defined asset job and retrying a run

### ISSUE THREAD

**Issue by esselius** [dagster-io/dagster#5578]
## Summary

I had created a software-defined asset job with `build_assets_job`, ran it, added a new asset and then tried to retry the job from a previous run snapshot and got this error

```
Operation name: RunRootQuery

Message: myDataFrame'

Path: ["pipelineRunOrError","executionPlan","steps",0,"inputs",0,"dependsOn",0,"outputs",0,"type"]

Locations: [{"line":51,"column":13},{"line":105,"column":11}]

Stack Trace:
  File "/usr/local/lib/python3.7/site-packages/graphql/execution/executor.py", line 452, in resolve_or_error
    return executor.execute(resolve_fn, source, info, **args)
  File "/usr/local/lib/python3.7/site-packages/graphql/execution/executors/sync.py", line 16, in execute
    return fn(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/dagster_graphql/schema/execution.py", line 36, in resolve_type
    return to_dagster_type(self._pipeline_snapshot, self._step_output_snap.dagster_type_key)
  File "/usr/local/lib/python3.7/site-packages/dagster_graphql/schema/dagster_types.py", line 26, in to_dagster_type
    dagster_type_key
  File "/usr/local/lib/python3.7/site-packages/dagster/core/snap/dagster_types.py", line 48, in get_dagster_type_snap
    return self.all_dagster_type_snaps_by_key[key]

```

## Additional Info about Your Environment

dagster 0.13.4 @ k8s deployed by helm

---
#### Message from the maintainers:
Impacted by this bug? Give it a 👍. We factor engagement into prioritization.

**Comment by yuhan** [comment]
Hi @esselius, thanks for reporting this issue. To better investigate the error, I'm going to ask some more specifics:
- was the error in Dagit?
- was the retry a re-execute all or a subset re-execution
- was the original run a full-graph execution? in other words, did you specify any op selection?

We've recently found a bug related to run snapshot with op selection and the fix went in 0.13.8. Wonder if upgrading to 0.13.8 and launching a brand new run still hit this error.

**Comment by esselius** [comment]
Hi @yuhan

yeah the error was in dagit with no op selection IIRC

I have upgraded to 0.13.8, but have cleared all state in that env since then, let's assume it was resolved or I will reopen :)

Thanks!
